lib/workers/repository/process/lookup/rollback.ts

```diff
@@ -38,12 +39,18 @@ export function getRollbackUpdate(
     'Versions found before rolling back'
   );
   lessThanVersions.sort((a, b) => version.sortVersions(a.version, b.version));
-  const newVersion = lessThanVersions.pop()?.version;
+  let newVersion = lessThanVersions.pop()?.version;
   // istanbul ignore if
   if (!newVersion) {
     logger.debug('No newVersion to roll back to');
     return null;
   }
+  while (
+    lessThanVersions.length !== 0 &&
+    !isVersionStable(newVersion, lessThanVersions, version)
+  ) {
+    newVersion = lessThanVersions.pop()?.version;
```

Comment:
And we probably should `ignoreUnstable` into account before filtering unstable versions.

Reply:
if we got to the while block and we couldnt find a stable version we will still return the last unstable vertion we checked. 
currently pop will never return `undefined` because of `lessThanVersions.length !== 0`.

thats why i suggest filtering the list if `ingoreUnstable == true` and then doing a single pop